### PR TITLE
Don't use xdist by default.

### PR DESCRIPTION
**Pull request by Drulex**
#### GitHub

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [x] I've linked relevant issues in the PR body
- [x] I've applied [the relevant labels](https://www.neuro.polymtl.ca/software/contributing#pr_labels) to this PR
- [x] I've assigned a reviewer

#### PR contents

- [x] I've consulted [SCT's internal developer documentation](https://github.com/neuropoly/spinalcordtoolbox/wiki) to ensure my contribution is in line with any relevant design decisions
- [ ] I've added [relevant tests](https://github.com/neuropoly/spinalcordtoolbox/wiki/Programming%3A-Tests) for my contribution
- [ ] I've updated the [relevant documentation](https://github.com/neuropoly/spinalcordtoolbox/wiki/Programming%3A-Documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

## Description
This PR removes the default usage of `pytest-xdist` when running the unit tests. See minutes from dev meeting:

- Motivation: we have hit some limitations/issues with regards to concurrency issues (https://github.com/pytest-dev/pytest-xdist/issues/271#issuecomment-763402552, https://github.com/neuropoly/spinalcordtoolbox/pull/3152). This complicates the ongoing effort to port CI to github actions (https://github.com/neuropoly/spinalcordtoolbox/pull/3125, https://github.com/neuropoly/spinalcordtoolbox/pull/3152)

- Other annoyances such as xdist disabling `-s` (`--capture=no`) thus preventing the user from viewing the output of `stdout` which is confusing for users. (Relevant dev Wiki page about this issue.)
   - This issue specifically is causing Julien to disable `xdist` for local testing, for example. 
- Doesn’t add value in CI since almost all VMs are single-core.
- Alternatives:
   - `pytest-parallel` (to be evaluated)
   - Disabled by default config. People who know what they are doing can enable it (or it can be enabled in ci.sh only).
   - Strategies in https://github.com/neuropoly/spinalcordtoolbox/issues/2674, particularly split tests across jobs; but we also need to consider that each job needs to spend installation time (there is a sweet spot in terms of number of tests per job)
- CI time on GH actions: total ~15min; on Travis: ~1hr
   - Largely dominated by waiting for the macOS VM to boot

In a nutshell: there is not a lot of value added (at the moment) in using `xdist` and working around the issues mentioned above. Tests can happily run sequentially and it won't make a difference for 99% of the people. People who do care about parallelization can still use `xdist` locally.

### CODE REVIEW

**Comment by Drulex** [comment]
TODO update wiki

**Comment by joshuacwnewton** [comment]
So, by removing `pytest-xdist`, we wouldn't need the locking fix from #3152, right? (cc'ing @kousu)

**Comment by Drulex** [comment]
> So, by removing `pytest-xdist`, we wouldn't need the locking fix from #3152, right? (cc'ing @kousu)

Indeed. @kousu may still contribute the fix upstream if he wants.

**Review by jcohenadad — APPROVED** [review]

**Comment by joshuacwnewton** [comment]
I've updated the Wiki here: https://github.com/neuropoly/spinalcordtoolbox/wiki/Contributing:-Testing-your-contribution#a-note-about-parallelization

**Comment by Drulex** [comment]
> I've updated the Wiki here: https://github.com/neuropoly/spinalcordtoolbox/wiki/Contributing:-Testing-your-contribution#a-note-about-parallelization

Thanks a lot!
